src/modules/producer.js

```diff
@@ -118,7 +119,7 @@ function checkRpc (queue, msg, options) {
 
         if (publishOrSendToQueue.call(this, queue, msg, options)) {
             //defered promise that will resolve when response is received
-            let responsePromise = Promise.defer();
+            let responsePromise =  new Deferred();
```

Comment:
I don't think we need another class just for this, and an Es6 one at that (we use var all over the place and have no engine requirement in package.json that would signal we need es6 support). Can't we just use a Promise or do that what you did here in an es5 way?